core/src/proof.rs

```diff
@@ -6,12 +6,30 @@ use crate::trie::{
 
 use bitvec::prelude::*;
 
+/// Wrapper for a terminal node, it will store the LeafData if it is a leaf node,
+/// and just the KeyPath to that terminal if it is a terminator node
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum PathProofTerminal {
+    Leaf(LeafData),
+    Terminator(KeyPath),
+}
+
+impl PathProofTerminal {
+    /// Return the KeyPath of the Terminal Node
+    pub fn key_path(&self) -> &[u8; 32] {
```

Comment:
likewise, with above, this can return a bit-slice but not a key-path, because terminators don't have full key-paths.